### ISSUE TITLE
Move block information out of the change trie.

### ISSUE THREAD

**Issue by cheme** [paritytech/substrate#3196]
This issue propose to remove "block" information from every change trie input keys.

Block number in change trie are mainly use for easy prunning.

This issue propose to use a prefix on every key in db that would be the encoded block number.

That way, pruning a block could be done by removing all keyvalue db that start with the encoded block number in the change trie rocksdb collection.

There will be no need to parse change trie content (currently we are collecting keys with proof building method).

Prefixing the key will be done with a PrefixedDB similar to 'KeySpacedDB' use in #2209 (with encoded block value instead of keyspace).
A variant of 'keyspace_as_prefix_alloc' from #2209 should also be use to avoid appending trie prefix.
Current 'MemoryDB' will also need to be replace with a 'PrefixedMemoryDB'.

Block number in key may still be needed for digest key.

So basically instead of having 'block' info in each trie key, 'block' info will be in each rocksdb key.

cc/ @svyatonik 
The main benefit here should be the speed of pruning, but it could also be interesting to see the effect of biggest rocksdb key instead of biggest trie key. Maybe it can end up being counterproductive.

**Comment by bkchr** [comment]
Changes trie was removed.
